File: javascript/json-transform/src/__tests__/functions/TransformerFunctionJsonPath.test.ts
import { describe, test } from "vitest";
import { assertTransformation } from "../BaseTransformationTest";
import { BigDecimal } from "../../functions/common/FunctionHelpers";

describe("TransformerFunctionJsonPath", () => {
  test("yoDawg", async () => {
    await assertTransformation(
      {
        path: "$.path",
      },
      "$$jsonpath($.path):$",
      "$.path",
    );
  });

  test("inArray", async () => {
    await assertTransformation(
      {
        arr: [null, "boo"],
      },
      "$$jsonpath('\\\\$.arr[1]'):$",
      "boo",
    );
  });

  test("multipleResults", async () => {
    await assertTransformation(
      [
        { id: 1, active: true },
        { id: 3, active: false },
        { id: 4, active: true },
        { id: 5, active: false },
      ],
      "$$jsonpath('\\\\$[?(@.active == true)]'):$",
      [
        { id: 1, active: true },
        { id: 4, active: true },
      ],
    );
  });
});

File: java/json-transform/src/test/java/co/nlighten/jsontransform/functions/TransformerFunctionJsonPathTest.java
package co.nlighten.jsontransform.functions;

import co.nlighten.jsontransform.BaseTest;
import org.junit.jupiter.api.Test;

public class TransformerFunctionJsonPathTest extends BaseTest {

    @Test
    void yoDawg() {
        assertTransformation(fromJson("""
        {
            "path": "$.path"
        }"""), "$$jsonpath($.path):$", "$.path");
    }

    @Test
    void inArray() {
        assertTransformation(fromJson("""
        {
            "arr": [
                null,
                "boo"
            ]
        }"""), "$$jsonpath('\\\\$.arr[1]'):$", "boo");
    }

    @Test
    void multipleResults() {
        assertTransformation(fromJson("""
        [
            { "id": 1, "active": true },
            { "id": 3, "active": false },
            { "id": 4, "active": true },
            { "id": 5, "active": false }
        ]"""), "$$jsonpath('\\\\$[?(@.active == true)]'):$", fromJson("""
        [
            { "id": 1, "active": true },
            { "id": 4, "active": true }
        ]"""));
    }
}
